arch/arm/core/aarch32/cortex_m/tz/secure_entry_functions.ld

```diff
@@ -5,20 +5,23 @@
  */
 
 /* nRF-specific defines. */
-#ifdef CONFIG_CPU_HAS_NRF_IDAU
+#if defined(CONFIG_CPU_HAS_NRF_IDAU) && CONFIG_ARM_NSC_REGION_BASE_ADDRESS == 0
 	/* This SOC needs the NSC region to be at the end of an SPU region. */
+	#define __NSC_ALIGN (ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE) \
+			- MAX(32, (1 << LOG2CEIL(__sg_size))))
 	#define NSC_ALIGN \
-		. = ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE) \
-			- (1 << LOG2CEIL(__sg_size))
+		. =  (__NSC_ALIGN + ((ABSOLUTE(.) > __NSC_ALIGN) \
+				? CONFIG_NRF_SPU_FLASH_REGION_SIZE : 0))
 	#define NSC_ALIGN_END . = ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE)
-#endif /* CONFIG_CPU_HAS_NRF_IDAU */
-
+#endif /* CONFIG_CPU_HAS_NRF_IDAU && CONFIG_ARM_NSC_REGION_BASE_ADDRESS != 0 */
 
+#ifndef NSC_ALIGN
 #if CONFIG_ARM_NSC_REGION_BASE_ADDRESS != 0
 	#define NSC_ALIGN . = ABSOLUTE(CONFIG_ARM_NSC_REGION_BASE_ADDRESS)
-#elif !defined(NSC_ALIGN)
+#else
 	#define NSC_ALIGN . = ALIGN(4)
```

Comment:
Why is this 4? Shouldn't it be rather 32 (TF-M uses such alignment for all platforms)?

Reply:
Does TrustZone-M require 32-byte alignment for Non-Secure callable functions/sections? If not, then the default 4-byte alignment should be fine...

Reply:
I've found the following requirement in "Arm®v8-M Security Extensions: Requirements on Development Tools":
> 13 A vector of secure gateway veneers must be aligned to a 32-byte boundary, and must be zero padded to a 32-byte boundary

In SAU, both `RBAR` and `RLAR` registers don't allow entering the lowest 5 address bits, those bits are defined to be always `0x00` and `0x1F`, respectively. That explains the above requirement. And I believe that all IDAUs will also require at least such alignment, but I don't have any document at hand to confirm this.

Reply:
Yes, this makes, sense, the RBAR and RLAR in SAU impose the same alignment restrictions as the same registers for MPU.

Reply:
However: in theory there would be IDAUs that do not have this requirement :);) Maybe we could do #elif SAUREGIONS_PRESENT != 0, instead of #else ?

Reply:
>in theory there would be IDAUs that do not have this requirement

IMO they should add their own ifdef if that is the case, like we have for our IDAU.

I added a commit increasing the start and end alignment to 32 bytes.

Reply:
OK